### ISSUE TITLE
Feedback?

### ISSUE THREAD

**Issue by karlhorky** [karlhorky/react-malarkey#2]
Since this is my first React component, I'd like to get some feedback as to how it could be improved.

This component is currently still in development, and thus there are some things that I already know I want to work on:
- api and setup documentation
- more demos with code examples
- visual presentation (css)
- more tests

@topa @jhnns @okonet any feedback you or anyone else can provide outside of these?

**Comment by okonet** [comment]
I've addded my feedback in the code. Check out my comments. Basically I think there is one big issue: your component will never react to `messages` change. So, if after mounting I'll pass new props, it will still render old ones. I didn't test myseelf, so probably I'm wrong. But from what I saw in code, it has this issue right now.

**Comment by jhnns** [comment]
> your component will never react to messages change

Yes, because it's part of `this.prop`. In order to re-render the component with new messages, `messages` should be part of `this.state` afaik. Thus you can trigger a re-render with `malarkey.setState(...)`

Why is `sequence` configurable? Wouldn't it be sufficient to just make the `forEach` callback configurable?

**Comment by karlhorky** [comment]
Thanks guys, I'll take a look at this tonight or tomorrow. :)
